src/utils.ml

```diff
@@ -205,6 +205,33 @@ module Ast_io = struct
         output_string oc Input_version.Ast.Config.ast_impl_magic_number;
         output_value oc input_name;
         output_value oc st
+
+    module Read_bin = struct
+    type ast =
+    | Intf of signature
+    | Impl of structure
+
+    type read_error' = Read_error of string
+
+    let ast_read_error_string (error : read_error) =
```

Comment:
As a side-note: you could call this function `read_error_to_string`. I think `<t1>_to_<t2>` is quite a common pattern to name a type conversion.

By the way, I'm wondering if it might make sense to move that function one module up to `Ast_io`, since that's where `read_error` is defined. What do you think about that?

Reply:
I agree with `read_error_to_string`that does make more sense as a name.

For your second point, I think it is indeed better to keep this function close to the type definition since keeping it in `Read_bin` module doesn't add any value
(and since i dont plan on exposing it because the error message is already passed though ` |  Error e -> Error (ast_read_error_string e)`.